src/domain/entities/Map.cpp

```diff
@@ -15,4 +17,11 @@ bool Map::is_transparent(Domain::Core::Position p) const
     const auto &tile = tiles_.at(p.x, p.y);
     return !tile.blocks_sight;
 }
+
+void Map::set_passable(Domain::Core::Position p, bool passable)
+{
+    if (!in_bounds(p))
+        return;
+    tiles_.at(p.x, p.y).blocks_movement = !passable;
+}
```

Comment:
Таке собі. Просто тихо повернути out of bounds... Не роби такого
```suggestion
[[nodiscard]] bool Map::set_passable(Position p, bool passable)
{
    if (!in_bounds(p)) return false;
    tiles_.at(p.x, p.y).blocks_movement = !passable;
    return true;
}
```